### PR TITLE
Feat/unified signup

### PR DESCRIPTION
**Pull request by TheLeoB**
## Description (optional)
Add a concise explanation of the changes.

## Testing Steps (optional)
Detail the steps required for the reviewer(s) to verify and test these changes.

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/layouts/Wallet/MenuLeft/Navigation/template.tsx [hunk 1]
```diff
@@ -136,11 +158,12 @@ const Navigation = (props: OwnProps & Props) => {
...
         </MenuItem>
       </LinkContainer>
-      <LinkContainer to='/exchange' activeClassName='active'>
-        <MenuItem data-e2e='exchangeLink'>
-          <ExchangeNavItem {...props} />
-        </MenuItem>
-      </LinkContainer>
+      <ExchangeMenuItem
+        data-e2e='exchangeLink'
+        onClick={() => props.profileActions.getExchangeLoginToken(ExchangeAuthOriginType.SideMenu)}
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/layouts/Wallet/MenuLeft/Navigation/template.tsx:163]: `authAndRouteToExchange`

File: packages/blockchain-wallet-v4-frontend/src/scenes/RecoverWallet/model.tsx [hunk 2]
```diff
@@ -191,11 +192,12 @@ export const BackArrowFormHeader = (props: {
           >
             {props.email}
           </Text>
-          {props.step !== RecoverSteps.RESET_ACCOUNT && (
-            <Text size='12px' weight={500} color='grey400'>
-              ({firstPartGuid}...{lastPartGuid})
-            </Text>
-          )}
+          {props.step !== RecoverSteps.RESET_ACCOUNT ||
+            (props.product === ProductAuthOptions.EXCHANGE && (
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/scenes/RecoverWallet/model.tsx:196]: should this be !Exchange?

File: packages/blockchain-wallet-v4/src/network/api/profile/index.ts [hunk 3]
```diff
@@ -47,6 +55,38 @@ export default ({ authorizedGet, authorizedPost, authorizedPut, nabuUrl, post, r
...
+      data: {
+        retailToken,
+        usersCredentialsId
+      },
+      endPoint: '/mercury/auth',
+      headers: {
+        Authorization: `Bearer ${exchangeLifetimeToken}`,
+        'X-CLIENT-TYPE': 'WEB',
+        'X-DEVICE-ID': null,
+        'x-app-version': '6.11.1'
```
Comment (schnogz) [inline packages/blockchain-wallet-v4/src/network/api/profile/index.ts:84]: double check what web should be sending here with Mr. P

Comment (schnogz) [reply packages/blockchain-wallet-v4/src/network/api/profile/index.ts:84]: check rest of codebase for some issue(?)

File: packages/blockchain-wallet-v4-frontend/src/data/auth/sagas.ts [hunk 4]
```diff
@@ -213,10 +264,18 @@ export default ({ api, coreSagas, networks }) => {
         yield put(actions.modules.settings.updateCurrency(currency, true))
         yield put(actions.core.settings.setCurrency(currency))
 
-        if (!isAccountReset) {
-          yield put(actions.router.push('/verify-email-step'))
+        if (isAccountReset) {
+          if (product === ProductAuthOptions.EXCHANGE) {
+            // yield put(actions.modules.profile.getExchangeLoginToken(ExchangeAuthOriginType.Login))
+            return
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/data/auth/sagas.ts:270]: double check this

**Review by schnogz — CHANGES_REQUESTED** [review]

File: packages/blockchain-wallet-v4/src/network/api/profile/index.ts (hunk 3 above)
Comment (TheLeoB) [reply packages/blockchain-wallet-v4/src/network/api/profile/index.ts:84]: only in one other place, checked with Pavel and their mandatory

**Review by schnogz — DISMISSED** [review]

**Review by schnogz — APPROVED** [review]

**Pull request merged**